base/collection_xattr_common.go

```diff
@@ -215,6 +215,9 @@ func WriteUpdateWithXattr(ctx context.Context, store *Collection, k string, xatt
 		if callbackExpiry != nil {
 			exp = *callbackExpiry
 		}
+		if updatedOpts != nil {
+			opts = updatedOpts
+		}
```

Comment:
If we assume the callback is doing something like "append some additional macro expansion options", then I feel like this block runs the risk of ending up with duplicates if this loop runs multiple times (cas mismatch on line 260).  Any reason we can't just leave opts intact, and pass updatedOpts to WriteWithXattr?

Reply:
This makes sense to me, will update to do this.